prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -4949,17 +4960,27 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
                             field.clone(),
                             location,
                             vir::AssignKind::Copy,
+                            true
                         )?;
-                        // Initialize the constant
-                        let const_val = self.encoder
-                            .encode_const_expr(ty, &val)
-                            .with_span(span)?;
-                        // Initialize value of lhs
-                        stmts.push(vir::Stmt::Assign(
-                            lhs.clone().field(field),
-                            const_val,
-                            vir::AssignKind::Copy,
-                        ));
+
+                        // TODO Encoding of string literals is not yet supported,
+                        // so do not encode an assignment if the RHS is a string
+                        let is_str = match ty.kind() {
+                            ty::TyKind::Ref(_, inner, _) => inner.is_str(),
+                            _ => false,
+                        };
+                        if !is_str {
```

Comment:
(For another PR) Could this be done for any type that `TypeEncoder::encode_predicate_def` encodes with a bodyless predicate, instead of just for `str`?

Reply:
I think that would be okay.